main/src/library/Map.flix

```diff
@@ -464,19 +464,38 @@ namespace Map {
     /// That is, the result is of the form: `f(v1, ...f(vn-1, f(vn, s)))`.
     /// A `foldRight` allows early termination by not calling the continuation.
     ///
-    pub def foldRight(f: (v, Unit -> b & ef) -> b & ef, z: b, m: Map[k, v]): b & ef =
-        foldRightWithKey((_, v, c) -> f(v, c), z, m)
+    pub def foldRight(f: (v, b) -> b & ef, z: b, m: Map[k, v]): b & ef =
+        foldRightWithKey((_, v, acc) -> f(v, acc), z, m)
 
     ///
     /// Applies `f` to a start value `s` and all key-value pairs in `m` going from right to left.
     ///
     /// That is, the result is of the form: `f(k1, v1, ...f(kn-1, vn-1, f(kn, vn, s)))`.
     /// A `foldRightWithKey` allows early termination by not calling the continuation.
     ///
-    pub def foldRightWithKey(f: (k, v, Unit -> b & ef) -> b & ef, z: b, m: Map[k, v]): b & ef =
+    pub def foldRightWithKey(f: (k, v, b) -> b & ef, z: b, m: Map[k, v]): b & ef =
         let Map(t) = m;
         RedBlackTree.foldRight(f, z, t)
 
+    ///
+    /// Applies `f` to a start value `s` and all values in `m` going from right to left.
+    ///
+    /// That is, the result is of the form: `f(v1, ...f(vn-1, f(vn, s)))`.
+    /// A `foldRightWithCont` allows early termination by not calling the continuation.
+    ///
+    pub def foldRightWithCont(f: (v, Unit -> b & ef) -> b & ef, z: b, m: Map[k, v]): b & ef =
+        foldRightWithKeyCont((_, v, c) -> f(v, c), z, m)
+
+    ///
+    /// Applies `f` to a start value `s` and all key-value pairs in `m` going from right to left.
+    ///
+    /// That is, the result is of the form: `f(k1, v1, ...f(kn-1, vn-1, f(kn, vn, s)))`.
+    /// A `foldRightWithKeyCont` allows early termination by not calling the continuation.
+    ///
+    pub def foldRightWithKeyCont(f: (k, v, Unit -> b & ef) -> b & ef, z: b, m: Map[k, v]): b & ef =
```

Comment:
@magnus-madsen I'm not sure, if this name `Map.foldrightWithKeyCont` is good? It is the `Map.foldRightWithKey` that calls `RedBlackTree.foldRightWithCont`.

Reply:
I think its OK.